### PR TITLE
New private method syntax guidelines

### PR DESCRIPTION
**Pull request by ChrisArcand**
**Updated:** After much discussion (about coding style? Imagine that), either convention for Ruby < 2.1 is acceptable; Ruby 2.1+ will use the `private def method_name` syntax allowed by Ruby 2.1 with `def` returning a symbol.

https://bugs.ruby-lang.org/issues/7998

-----

After using the `private :method` with its definition for some time in internal Sport Ngin code and using the [more standard] singular `private` declaration in _everything else in the Ruby world_, I argue that the former does more harm than good and is a very awkward non-conventional-looking Ruby.

The current convention...
* Is not a convention. I have yet to see a single project which _consistently_ uses `private :method`. In fact, if you work with any of the major open source Ruby projects you will find that this proposed change is essentially how Ruby is idiomatically written. No, really - if anyone knows of any major projects that use the current style, ping me, I'd like to see it.
* Adds an insane amount of clutter to files. If you have a larger class, it's not actually easier to read with private declarations all over - it's quite a bit harder.
* Allows you to mix public/private/projected methods, allowing for more mistakes if you forget the private declaration (happens all the time, I've seen it).
* I literally took the code in this PR from the very [styleguide that this one is based on.](https://github.com/bbatsov/ruby-style-guide/blob/master/README.md#indent-public-private-protected)

Misc responses to arguments for the current style that I've heard:

> But then declaration of `private` is always found with the method; when you have a super long class, you can't get lost in what is and isn't private!

* If you have a super long class where you literally get lost in method definitions, that's code smell. You've got much bigger problems, my friend.
* If that super long class already exists and you can't really refactor it (you just have to deal with it), it's still going to look more awful and cluttered using the current style.
* If you are adding a method to the bottom of a class that you intend to be public, I'm happy with that - when Ruby yells at you that you've called a private method and need to think harder about where you're placing it, that's a good thing.

> Where the method is located in a file doesn't have any bearing on it's privacy

* Who says that's a bad thing? If you used _standard convention_, you'd be knocking out two styleguide points at once, as you'd be [essentially] forced to put private methods at the bottom of the file where they belong.

Summing up, let's write Ruby how Ruby is usually written (it's done that way for a reason).~~

### CODE REVIEW

**Comment by carlallen** [comment]
I can't remember which project it was that I found using this standard, but when I found it, it was awesome.

**Comment by mpillsbury** [comment]
https://github.com/sportngin/auto_mobile uses `private :method` consistently. (Because of this style guide.)

**Comment by anfleene** [comment]
good for you
